samples/subsys/usb/cdc_acm/src/main.c

```diff
@@ -41,12 +41,27 @@ static void interrupt_handler(struct device *dev)
 
 static void write_data(struct device *dev, const char *buf, int len)
 {
+	int sent;
+	u32_t transfer_len;
 	uart_irq_tx_enable(dev);
 
-	data_transmitted = false;
-	uart_fifo_fill(dev, (const u8_t *)buf, len);
-	while (data_transmitted == false)
-		;
+	while (len) {
+		data_transmitted = false;
+		/* FIXME: An issue observed when USB TX Endpoint is written
+		 * more than once (greater than 4 Bytes of data is written)
+		 * (Jira 2074). So restricting writing 4 bytes at a time into
+		 * USB TX Endpoing.
+		 */
+		transfer_len = len > sizeof(u32_t) ? sizeof(u32_t) : len;
```

Comment:
The code to limit transfer size is in the wrong place. If cdc_acm_fifo_fill() is known to break when the argument is >4 then this is bug in cdc_acm_fifo_fill() and any fix/mitigation/workaround must go there.

Having this sort of code and putting in an example will simply propagate bad practice throughout the zephyr ecosystem.